agent/dns.go

```diff
@@ -1080,14 +1073,38 @@ func dnsBinaryTruncate(resp *dns.Msg, maxSize int, index map[string]dns.RR, hasE
 // trimTCPResponse limit the MaximumSize of messages to 64k as it is the limit
 // of DNS responses
 func trimTCPResponse(req, resp *dns.Msg) (trimmed bool) {
-	hasExtra := len(resp.Extra) > 0
-	// There is some overhead, 65535 does not work
+	// NOTE: 65523 is a magic number that has been changed several times in the
```

Comment:
Note to reviewer: I did not attempt to fix the magic numbers, but I did want to describe them so that someone with more DNS domain knowledge would have an easier time fixing these numbers in the future.